pandas/core/frame.py

```diff
@@ -835,15 +835,81 @@ def __len__(self):
 
     def dot(self, other):
         """
-        Matrix multiplication with DataFrame or Series objects
+        Find the dot product of two DataFrames.
+
+        Using two DataFrame objects - this method
+        finds the standard inner product (dot product)
+        of two conformable matrices. Matrices A and B are said
+        to be conformable for multiplication A(B) whenever the
+        number of columns in A = the number of rows in B.
+        (i.e. -  A is m × p and B is p × n)
 
         Parameters
         ----------
-        other : DataFrame or Series
+        other : DataFrame
+            A DataFrame of the correct dimension.
 
         Returns
         -------
-        dot_product : DataFrame or Series
+        dot_product : DataFrame
+
+        Raises
+        ------
+        ValueError
+            If the first dimension of A is not the same size as the
+            second dimension of B, OR if there are conflicting column
+            indeces.
+
+        See Also
+        --------
+        numpy.dot : dot product in numpy
+
```

Comment:
Maybe include a reference to `Series.dot` as well (since it is used in the examples)?